hs-abci-extra/src/Network/ABCI/Server/Middleware/MetricsLogger.hs

```diff
@@ -0,0 +1,145 @@
+module Network.ABCI.Server.Middleware.MetricsLogger
+    ( -- * Basic stdout logging
+      mkMetricsLogStdout
+    --, mkMetricsLogDatadog
+    , mkMetricsLogDatadogLocal
+      -- * Custom Loggers
+    , mkMetricsLogger
+    , mkMetricsLoggerM
+    ) where
+import           Control.Concurrent.MVar   (MVar, modifyMVar, newMVar)
+import           Control.Monad.IO.Class    (MonadIO, liftIO)
+import qualified Data.Aeson                as A
+import           Data.Map.Strict           (Map)
+import qualified Data.Map.Strict           as Map
+import           Data.Time                 (NominalDiffTime, diffUTCTime,
+                                            getCurrentTime)
+import           Katip
+import           Katip.Scribes.Datadog.TCP
+import           Network.ABCI.Server.App   (App (..), MessageType (..),
+                                            Middleware, Request (..),
+                                            msgTypeKey)
+import           System.IO                 (stdout)
+
+---------------------------------------------------------------------------
+-- mkMetricsLogStdout
+---------------------------------------------------------------------------
+-- | Creates a production Metrics logger as middleware for ABCI Server.
+-- Uses lowest possible verbosity, however, with the current minimal metrics
+-- verbosity has no effect on which metrics logged.
+mkMetricsLogStdout :: (MonadIO m) => m (Middleware m)
+mkMetricsLogStdout = do
+  handleScribe <- liftIO $ mkHandleScribe ColorIfTerminal stdout (permitItem InfoS) V0
+  le <- liftIO (registerScribe "stdout" handleScribe defaultScribeSettings
+        =<< initLogEnv "ABCI" "production")
+  let ns = "Server"
+  mvarReqC <- liftIO $ newMVar Map.empty
+  pure $ mkMetricsLogger mvarReqC le ns
+
+---------------------------------------------------------------------------
+-- mkMetricsLogDatadog
+---------------------------------------------------------------------------
+
+mkMetricsLogDatadogLocal :: (MonadIO m) => Integer -> m (Middleware m)
+mkMetricsLogDatadogLocal port = do
+  datadogScribeSettings <- liftIO $ mkDatadogScribeSettings (localAgentConnectionParams (fromInteger port)) NoAuthLocal
+  scribe <- liftIO $ mkDatadogScribe datadogScribeSettings (permitItem InfoS) V0
+  le <- liftIO (registerScribe "datadog" scribe defaultScribeSettings
+        =<< initLogEnv "ABCI" "production")
+  let ns = "Server"
+  mvarReqC <- liftIO $ newMVar Map.empty
+  pure $ mkMetricsLogger mvarReqC le ns
+
+-- mkMetricsLogDatadog :: (MonadIO m) => m (Middleware m)
+-- mkMetricsLogDatadog = do
+--   let apiKey = APIKey undefined
```

Comment:
Unsure how to address this. Maybe an env var?

Reply:
why not take it as an arg and use an environment variable whenever it gets used in application code to create this arg